### ISSUE TITLE
[Enhancement]: Immediately insert new word via the context menu option when a new word is selected.

### ISSUE THREAD

**Issue by GilShoshan94** [streetsidesoftware/vscode-spell-checker#1049]
Hi,

I already explained it in #1048.

I suggest that if a user is with the cursor on an unknown word, the word should be added immediately and there is no need for a confirmation via the palette.

So this is the case:
![image](https://user-images.githubusercontent.com/29143941/125775557-5b49ef87-e4b3-48d0-831f-b04037a80391.png)

And this is the confirmation in the palette that I think should be skipped.
![image](https://user-images.githubusercontent.com/29143941/125775575-aa209173-a339-4e38-b23b-be8680e5a029.png)

**Comment by Jason3S** [comment]
Please try out v2 of the spell checker: [Releases · vscode-spell-checker](https://github.com/streetsidesoftware/vscode-spell-checker/releases). The right click menu and `ctrl .` options have changed. Your feedback would be welcome.

**Comment by GilShoshan94** [comment]
@Jason3S 

I just install the 2.0.1 version you asked me too.

I tried the right click (now I am using only the `cspell.json` file, without a separated custom dictionary txt file)

I see the new context menu, great job 👍 !

![image](https://user-images.githubusercontent.com/29143941/125790534-a146ed6f-5a74-4f33-acc4-bf7932253d8b.png)

But I still see strange behaviors.

**Comment by GilShoshan94** [comment]
Here, my `.vscode/cspell.json` :
```jsonc
{
    "version": "0.1",
    "allowCompoundWords": true,
    "minWordLength": 4,
    "spellCheckDelayMs": 100,
    "ignorePaths": [
        "**/target/debug/**",
        "**/target/release/**",
        "**/target/*.TAG"
    ],
    "words": [
        "gil",
        "shoshan"
    ]
}
```

here my code file `src/main.rs`:
```rust
// Author Gil Shoshan
fn main() {
    println!("Hello, world!");
    println!("New word 1: Creati, 2: Zsdaf");
}
```

Now I get `Creati` and `Zsdaf` corectly flagged (and my name, `Shoshan` passed well):
![image](https://user-images.githubusercontent.com/29143941/125791211-4999cddc-6a89-4149-beff-c81b4193cf12.png)

Then, I put my cursor and my mouse on `Creati`, right-click, in the context menu I select `Add Word to CSpell Configuration`.
And now my `.vscode/cspell.json` changed a looked like that:
![image](https://user-images.githubusercontent.com/29143941/125791564-cba7c0ea-8eeb-467b-a61b-ca7a9042e7b9.png)

So I expected 2 things:
- I expected only the word I was selecting to be inserted into the list, instead I got both of them (so I am guessing it add all the unknow word). This is an issue.
- I was expecting the order to be first all the capital letters than the lower cap in the list of words, based on my understanding from the previous version. But I guess you change the sorting order ? If the order is alphanumeric sort where the 'a' < 'B', and 'A' < 'a', I think it's a good idea and prefer that then a ASCII value based sort where 'B' < 'a'.

**Comment by GilShoshan94** [comment]
I reset my test case (explained in my previous comment) and I tried with `ctrl .`:

![image](https://user-images.githubusercontent.com/29143941/125792512-4932b4a7-ea71-4e7a-ae1f-4cc6cd82bc34.png)

And this worked perfectly as intended 👍 !

Inserted one by one, and the sorting order seems consistent to what I am guessing.

**Comment by GilShoshan94** [comment]
@Jason3S 

I experimented more with the right-click context menu, and we have issue there.
So I reset my test.

When I right click in the background anywhere on the editor, and select `Add Word to CSpell Configuration`, I was expecting to get the command palette to open and to let me input my new word (as in the previous version).
Instead, all the flagged words gets added in my `cspell.json` words list.

Also, by curiosity I tried the other options some are not implemented yet, I am sure you know. but my issued was with  `Add Word to User Configuration`. When I clicked that, again, no command palette appears, but also all the flagged words gets added in my `cspell.json` words list, which is not the "user" space. I was expected them to be add in the user dictionary I set up or in the user `settings.json` in the `"cSpell.userWords": []` list.

**Comment by Jason3S** [comment]
@GilShoshan94,

Thank you for the input.

The right click menu seems to be a bit flaky at the moment. Sorry about that.

I'm actively trying to improve the context menu and the options. The challenge is that an extension has very little control over the context menu. All the items in the context menu must be defined at development time, not run time.

The spell checker tries to determine the best set of targets for the words you want to add without listing ALL the possible options.

In any case, I just tried the right click menu, and it is not working as expected. I many have messed up the targets while adjusting the commands.

**Comment by Jason3S** [comment]
@GilShoshan94,

Many of the right click options should be fixed. Please try them out: 
[Release Release v2.0.1-alpha.7 · streetsidesoftware/vscode-spell-checker](https://github.com/streetsidesoftware/vscode-spell-checker/releases/tag/v2.0.1-alpha.7)

**Comment by GilShoshan94** [comment]
@Jason3S.

I tried today. 
I like the addition of `Spelling Suggestion...`

Still some issues:

So this is my `cspell.json`:
```json
{
    "version": "0.1",
    "allowCompoundWords": true,
    "minWordLength": 4,
    "spellCheckDelayMs": 100,
    "ignorePaths": [
        "**/target/debug/**",
        "**/target/release/**",
        "**/target/*.TAG"
    ],
    "words": [
        "gil",
        "shoshan"
    ]
}
```

And when I add the word `Mycustomword` by clicking ``Add Word to CSpell Configuration` in the context menu, it changes it to this:

```json
{

    "version": "0.1",

    "allowCompoundWords": true,

    "minWordLength": 4,

    "spellCheckDelayMs": 100,

    "ignorePaths": [

    
    "**/target/debug/**",

    
    "**/target/release/**",

    
    "**/target/*.TAG"

    ],

    "words": [

    
    "gil",

    
    "Mycustomword",

    
    "shoshan"

    ]
}
```


And the `Add Word to User Settings` didn't work for me. I got:
![image](https://user-images.githubusercontent.com/29143941/126213302-0c35b594-eec4-4762-90b6-9a2d07b4e199.png)

And the `Add Word to Workspace Settings` did work the first time I tried it. It generated a `settings.json` in the `.vscode` folder (because I didn't had one before) and correctly filled the word as expected. I then deleted this `settings.json` file and when I retry the `Add Word to Workspace Settings`, I got the same error than with the `Add Word to User Settings`.

**Comment by GilShoshan94** [comment]
So the modified `cspell.json` is correct but it add extra newlines that I think should not.

**Comment by Jason3S** [comment]
@GilShoshan94,

Great catch on the formatting issue. It was caused by the code trying to preserve the indentation. This should be fixed.

Add to Folder/Workspace/User settings should also be fixed.

[Release v2.0.1-alpha](https://github.com/streetsidesoftware/vscode-spell-checker/releases)

**Comment by Jason3S** [comment]
This should be fixed with the latest release: [Release v2.0.1-alpha](https://github.com/streetsidesoftware/vscode-spell-checker/releases).
